File: dp_labs_and_solutions/dp_labs_and_solutions/dp_java_labs_and_solutions/interpreter/lab/Divide.java
// Divide.java

import java.io.PrintWriter;

public class Divide extends Binary
{
	public Divide(Expression aOperand1, Expression aOperand2)
	{
		super(aOperand1, aOperand2);
	}

	public void output(PrintWriter aPrintWriter)
	{
		outputInfix(aPrintWriter, "/");
	}
}

File: dp_labs_and_solutions/original/dp_labs_and_solutions/dp_cpp_labs_and_solutions/interpreter/lab/divide.cpp
// divide.cpp

#include <iostream>
#include "binary.h"
#include "divide.h"
#include "dp_exception.h"
class expression;

divide::divide(expression * a_operand1, expression * a_operand2)
: binary(a_operand1, a_operand2)
{
}

void divide::output(std::ostream & a_ostream) const
{
	output_infix(a_ostream, "/");
}
